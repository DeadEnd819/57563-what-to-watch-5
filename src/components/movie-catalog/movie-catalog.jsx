import React, {memo} from "react";
import {connect} from "react-redux";
import {changeGenre, incrementShowFilmsCount} from "../../store/action";
import PropTypes from "prop-types";
import {FilmCardType} from "../../prop-types/prop-types";
import MovieGenresList from "../movie-genres-list/movie-genres-list";
import ShowMoreButton from "../show-more-button/show-more-button";
import MovieList from "../movie-list/movie-list";
import {getGenre, getGenres, getFilmsCount, getFilteredFilms} from "../../store/selectors";

const MemoMovieGenresList = memo(MovieGenresList);

const MovieCatalog = ({genresList, filteredFilms, genre, showFilmsCount, changeGenreAction, showMoreCards}) => {
  const handleButtonClick = (evt) => {
    evt.preventDefault();
    const currentGenre = evt.target.textContent;
    changeGenreAction(currentGenre);
  };

  return (
    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>

      <MemoMovieGenresList genresList={genresList} activeGenre={genre} onGenreClick={handleButtonClick} />

      <MovieList films={filteredFilms} showCount={showFilmsCount} />

      {filteredFilms.length > showFilmsCount ? <ShowMoreButton onShowMoreClick={showMoreCards} /> : ``}
    </section>
  );
};

MovieCatalog.propTypes = {
  genresList: PropTypes.arrayOf(PropTypes.string).isRequired,
  filteredFilms: PropTypes.arrayOf(FilmCardType).isRequired,
  genre: PropTypes.string.isRequired,
  showFilmsCount: PropTypes.number.isRequired,
  changeGenreAction: PropTypes.func.isRequired,
  showMoreCards: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  genre: getGenre(state),
  genresList: getGenres(state),
  showFilmsCount: getFilmsCount(state),
  filteredFilms: getFilteredFilms(state),
});

const mapDispatchToProps = (dispatch) => ({
  changeGenreAction(genre) {
    dispatch(changeGenre(genre));
  },
  showMoreCards() {
    dispatch(incrementShowFilmsCount());
  },
});

export {MovieCatalog};
export default connect(mapStateToProps, mapDispatchToProps)(MovieCatalog);
