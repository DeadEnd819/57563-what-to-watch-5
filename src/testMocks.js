export const promoFilm = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  genre: `Comedy`,
  released: 2014,
  isFavorite: false,
};

export const film = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  previewImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: true,
};

export const reviewOne = {
  id: 1,
  user: {
    id: 4,
    name: `Kate Muir`,
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
};

export const reviewTwo = {
  id: 2,
  user: {
    id: 5,
    name: `Paula Fleri-Soler`,
  },
  rating: 9.1,
  comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  date: `2019-08-08T17:40:56.569Z`,
};

export const reviews = [reviewOne, reviewTwo];

export const genresList = [
  `Action`,
  `Adventure`,
  `Comedy`,
  `Crime`,
  `Drama`,
  `Fantasy`,
  `Thriller`
];

export const state = {
  DATA: {
    films: [film, film, film, film, film, film, film],
    promo: promoFilm,
    currentFilm: film,
    currentReviews: reviews,
  },
  FILTER: {
    genre: `Drama`,
    showFilmsCount: 8,
  },
  USER: {
    name: `name`,
    avatar: `avatar.jpg`,
    status: `NOT_AUTHORIZED`,
    reviewStatus: `REVIEW_NOT_UPDATED`,
  },
};
