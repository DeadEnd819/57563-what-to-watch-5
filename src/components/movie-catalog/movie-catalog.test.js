import React from "react";
import renderer from "react-test-renderer";
import {MovieCatalog} from "./movie-catalog";
import {genresList, film} from "../../testMocks";

jest.mock(`../movie-genres-list/movie-genres-list.jsx`, () => `MovieGenresList`);
jest.mock(`../show-more-button/show-more-button.jsx`, () => `ShowMoreButton`);
jest.mock(`../movie-list/movie-list.jsx`, () => `MovieList`);

const noop = ()=>{};

describe(`Render MovieCatalog`, () => {
  it(`Should MovieCatalog 8 films render correctly`, () => {
    const tree = renderer
      .create(
          <MovieCatalog
            genresList={genresList}
            filteredFilms={[film, film, film, film, film, film, film, film, film, film]}
            genre={`Action`}
            showFilmsCount={8}
            changeGenreAction={noop}
            showMoreCards={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should MovieCatalog 16 films render correctly`, () => {
    const tree = renderer
      .create(
          <MovieCatalog
            genresList={genresList}
            filteredFilms={[film, film, film, film, film, film, film, film, film, film]}
            genre={`Action`}
            showFilmsCount={16}
            changeGenreAction={noop}
            showMoreCards={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
