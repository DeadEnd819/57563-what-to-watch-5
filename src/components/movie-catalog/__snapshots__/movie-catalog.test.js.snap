// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCatalog Should MovieCatalog 8 films render correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <MovieGenresList
    activeGenre="Action"
    genresList={
      Array [
        "Action",
        "Adventure",
        "Comedy",
        "Crime",
        "Drama",
        "Fantasy",
        "Thriller",
      ]
    }
    onGenreClick={[Function]}
  />
  <MovieList
    films={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
      ]
    }
    showCount={8}
  />
  <ShowMoreButton
    onShowMoreClick={[Function]}
  />
</section>
`;

exports[`Render MovieCatalog Should MovieCatalog 16 films render correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <MovieGenresList
    activeGenre="Action"
    genresList={
      Array [
        "Action",
        "Adventure",
        "Comedy",
        "Crime",
        "Drama",
        "Fantasy",
        "Thriller",
      ]
    }
    onGenreClick={[Function]}
  />
  <MovieList
    films={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
        Object {
          "backgroundColor": "#ffffff",
          "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
          "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
          "director": "Wes Andreson",
          "genre": "Comedy",
          "id": 1,
          "isFavorite": true,
          "name": "The Grand Budapest Hotel",
          "posterImage": "the-grand-budapest-hotel-poster.jpg",
          "previewImage": "bg-the-grand-budapest-hotel.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 8.9,
          "released": 2014,
          "runTime": 99,
          "scoresCount": 240,
          "starring": Array [
            "Bill Murray",
            "Edward Norton",
            "Jude Law",
            "Willem Dafoe",
            "Saoirse Ronan",
          ],
        },
      ]
    }
    showCount={16}
  />
  
</section>
`;
